src/platform.h

```diff
@@ -92,16 +96,19 @@ namespace Platform {
 		int64_t GetSize() const;
 
 	private:
-		const std::string filename;
 #ifdef _WIN32
-		std::wstring filename_w;
+		const std::wstring filename;
+#else
+		const std::string& filename;
```

Comment:
This is really easy to get wrong. I can write the following:

```
File f("Hello World");
```

It will compile but cause a dangling reference error at runtime. The `const std::string&` constructor accepts temporaries which will get destroyed immediately.

Unfortunately there isn't a good way to do this that gives you both an easy to use interface and maximum efficiency (Unless you make a static `Exists()` method which just takes in a string, but then you also need to do windows `w_string` conversion).

Given this is only used internally for FileFinder, I'm fine with either approach for now.

Reply:
Okay didn't knew about this corner case. Will keep the copy then. As "Exists" is not really in a hot codepath (just for rtp init and project init).

Reply:
Agree, for now I'd rather keep the copy than have an interface like this.